### PR TITLE
Feature/datasource ess notifications

### PR DESCRIPTION
**Pull request by juhuahaoteng**
IT-C1MRWWDYH3QD:terraform-provider-alicloud chore$ TF_ACC=1 go test ./alicloud -v -run=TestAccAlicloudEssNotificationsDataSource -timeout=300m
=== RUN   TestAccAlicloudEssNotificationsDataSource
--- PASS: TestAccAlicloudEssNotificationsDataSource (58.38s)
PASS
ok      github.com/terraform-providers/terraform-provider-alicloud/alicloud     (cached)

### CODE REVIEW

File: alicloud/data_source_alicloud_ess_notifications.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+			return WrapErrorf(err, DataDefaultErrorMsg, "alicloud_ess_notifications", request.GetActionName(), AlibabaCloudSdkGoERROR)
+		}
+		addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+		response := raw.(*ess.DescribeNotificationConfigurationsResponse)
+		if len(response.NotificationConfigurationModels.NotificationConfigurationModel) < 1 {
+			break
+		}
+		allNotifications = append(allNotifications, response.NotificationConfigurationModels.NotificationConfigurationModel...)
+		if len(response.NotificationConfigurationModels.NotificationConfigurationModel) < PageSizeLarge {
+			break
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_ess_notifications.go]: else?

File: alicloud/data_source_alicloud_ess_notifications.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+		ids = append(ids, n.ScalingGroupId)
+		s = append(s, mapping)
+	}
+	d.SetId(dataResourceIdHash(ids))
+
+	if err := d.Set("notifications", s); err != nil {
+		return WrapError(err)
+	}
+
+	if err := d.Set("ids", ids); err != nil {
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_ess_notifications.go]: If set ids, please add it in the schema

File: website/docs/d/ess_notifications.html.markdown [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+page_title: "Alicloud: alicloud_ess_notifications"
+sidebar_current: "docs-alicloud_ess_notifications"
+description: |-
+    Provides a list of notifications available to the user.
+---
+
+# alicloud_ess_scaling_rules
+
+This data source provides available notification resources. 
+
```
Comment (xiaozhu36) [inline website/docs/d/ess_notifications.html.markdown]: please add note

File: alicloud/data_source_alicloud_ess_notifications.go [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+		response := raw.(*ess.DescribeNotificationConfigurationsResponse)
+		if len(response.NotificationConfigurationModels.NotificationConfigurationModel) < 1 {
+			break
+		}
+		allNotifications = append(allNotifications, response.NotificationConfigurationModels.NotificationConfigurationModel...)
+		if len(response.NotificationConfigurationModels.NotificationConfigurationModel) < PageSizeLarge {
+			break
+		} else {
+			continue
+		}
```
Comment (Lexsss) [inline alicloud/data_source_alicloud_ess_notifications.go]: ids filter

File: alicloud/data_source_alicloud_ess_notifications_test.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+	"github.com/hashicorp/terraform/helper/acctest"
+)
+
+func TestAccAlicloudEssNotificationsDataSource(t *testing.T) {
+	rand := acctest.RandInt()
+	allConf := dataSourceTestAccConfig{
+		existConfig: testAccCheckAlicloudEssNotificationsDataSourceConfig(rand, map[string]string{
+			"scaling_group_id": `"${alicloud_ess_notification.default.scaling_group_id}"`,
+			"ids":              `["${alicloud_ess_notification.default.id}"]`,
```
Comment (Lexsss) [inline alicloud/data_source_alicloud_ess_notifications_test.go]: no filter logic

File: website/docs/d/ess_notifications.html.markdown [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+The following arguments are supported:
+
+* `scaling_group_id` - (Required) Scaling group id the notifications belong to.
+* `output_file` - (Optional) File name where to save data source results (after running `terraform plan`).
+
+## Attributes Reference
+
+The following attributes are exported in addition to the arguments listed above:
+
+* `ids` - A list of notification ids.
```
Comment (Lexsss) [inline website/docs/d/ess_notifications.html.markdown]: move to Argument

File: alicloud/data_source_alicloud_ess_notifications_test.go [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+)
+
+func TestAccAlicloudEssNotificationsDataSource(t *testing.T) {
+	rand := acctest.RandInt()
+	allConf := dataSourceTestAccConfig{
```
Comment (Lexsss) [inline alicloud/data_source_alicloud_ess_notifications_test.go]: scalingGroupIdConf

**Pull request merged**